class/gitreminder.class.php

```diff
@@ -48,76 +69,396 @@ public function __construct()
     {
     	if (!file_exists($file))
     	{
     		throw new Exception("File '$file' not found!",404);
+            $this->log->warning("File not found: $file");
     	}
     	
-    	array_push($this->tasks, unserialize(file_get_contents($file)));
-    	    	
+    	$this->tasks = array_merge($this->tasks, unserialize(file_get_contents($file)));
+    
         return $this;
     }
-
+    
+    
     /**
-     * Load unread GitHub-Notifications and store them to
-     * $this->tasks[]
+     * Load stored task from Database
+     * @param string $dbHost
+     * @param string $dbUser
+     * @param string $dbName
+     * @param string $dbPwd
+     */
+    public function loadStoredTasksFromDatabase()
+    {
+    	throw new \Exception ('Protocol "$type" not implemented yet', 501);
+    }
+    
+    
+    
+    /**
+     * Load encoded tasks from last run
      *
-     * @todo Implement
+     * @param $jfile
      * @return $this
      */
-    public function loadGhNotifications()
+    public function loadStoredTasksJson($jFile = self::FILE_TASKS_JSON)
     {
-        return $this;
+    	if (!file_exists($file))
+    	{
+    		throw new Exception("File '$jfile' not found!",404);
+            $this->log->warning("File not found: $jFile");
+    	}
+    	 
+    	array_push($this->tasks,json_decode(file_get_contents($jFile)));
+    
+    	return $this;
     }
 
+    
+    
+    
+    
+    /**
+     * Load unread GitHub-Notifications and store them to $this->tasks[]
+     * @param string $nameGitReminder
+     * @return $this
+     */
+    public function loadGhNotifications($nameGitReminder = self::NAME_OF_GITREMINDER)
+    {
+    	// @TODO: Find methode to read current notifications
+    	// @TODO: Don't load unnecessary api-urls to make the json smaller
+    	// This is quite ugly, but it's the only way we found to get access to
+    	// a list of all notifications including the html_url and fe_issue_id
+    	// all methodes within githubRepo->activity->notifications->listYourNotifications()[x] 
+    	// are protected so we could not load them from our position.
+  	
+    	
+    	//We are looking for new notifications and return them as an Array in var $notification
+    	$notifications = json_decode($this->githubRepo->request("/notifications", 'GET', array('participating' => true), 200, 'string', true), true);
+    	
+    	//var_dump($notifications);
+    	
+        if(count($notifications)>=30)$this->log->warning("$name$nameGitReminder has been called more than 30 times\n\n",$notifications);
+        
+    	foreach ($notifications as $element)
+    	{
+    		$repoOwner = $element["repository"]["owner"]["login"];
+    		$repo =  $element["repository"]["name"];
+    		$issueTitel = $element["subject"]["title"];
+    		$issue_path_ok = str_replace("https://api.github.com","",$element["subject"]["url"]);
+    		$issueId = intval(str_replace("/repos/$repoOwner/$repo/issues/","",$issue_path_ok));

```

Comment:
This can't be the best way!
I don't even think you need the `$issueId` in your code without the url.
otherwise u will find a way to export the id a better way